### PR TITLE
Update ADBC to 0.7.0

### PR DESCRIPTION
**Pull request by josevalim**
@superhawk610, unfortunately the updating is failing on PG tests for me. Did you run into the same issue? If you update it, do you see the same?

### CODE REVIEW

**Comment by superhawk610** [comment]
Postgres appears to work correctly for me, I'm just seeing these two errors from the SQLite tests

```elixir
  1) test query select with parameters (Adbc.Connection.Test)
     test/adbc_connection_test.exs:178
     match (=) failed
     code:  assert {:ok, %Adbc.Result{data: %{"num" => [579]}}} =
              Connection.query(conn, "SELECT 123 + ? as num", [456])
     left:  {:ok, %Adbc.Result{data: %{"num" => [579]}}}
     right: {
              :error,
              %Adbc.Error{
                __exception__: true,
                message: "Failed to initialize array view: (22) Invalid argument: Expected struct array buffer 0 to have size >= 1 bytes but found buffer with 0 bytes",
                state: <<0, 0, 0, 0, 0>>,
                vendor_code: 0
              }
            }
     stacktrace:
       test/adbc_connection_test.exs:181: (test)



  2) test query! select with parameters (Adbc.Connection.Test)
     test/adbc_connection_test.exs:203
     ** (Adbc.Error) Failed to initialize array view: (22) Invalid argument: Expected struct array buffer 0 to have size >= 1 bytes but found buffer with 0 bytes
     code: Connection.query!(conn, "SELECT 123 + ? as num", [456])
     stacktrace:
       (adbc 0.1.1) lib/adbc_connection.ex:82: Adbc.Connection.query!/3
       test/adbc_connection_test.exs:207: (test)
```

**Comment by josevalim** [comment]
Yeah, sorry, I spoke incorrectly. I think we are encoding parameters incorrectly or they changed the structure.

File: 3rd_party/apache-arrow-adbc/adbc.h [hunk 1]
```diff
@@ -35,7 +35,7 @@
 /// but not concurrent access.  Specific implementations may permit
 /// multiple threads.
 ///
-/// \version 1.0.0
```
Comment (superhawk610) [inline 3rd_party/apache-arrow-adbc/adbc.h:38]: Looks like the new Arrow API version may be the cause 🤔

**Review by superhawk610 — COMMENTED** [review]
This is the source of the `Failed to initialize array view` error:

https://github.com/elixir-explorer/adbc/blob/293bca044f55fe3cffde30d50b4f16435964f79c/3rd_party/apache-arrow-adbc/c/driver/sqlite/statement_reader.c#L283-288

File: c_src/adbc_nif.cpp [hunk 2]
```diff
@@ -1249,13 +1249,9 @@ int elixir_to_arrow_type_struct(ErlNifEnv *env, ERL_NIF_TERM values, struct Arro
 
     ArrowSchemaInit(schema_out);
     NANOARROW_RETURN_NOT_OK(ArrowSchemaSetTypeStruct(schema_out, n_items));
-    NANOARROW_RETURN_NOT_OK(ArrowSchemaSetName(schema_out, ""));
-
     NANOARROW_RETURN_NOT_OK(ArrowArrayInitFromType(array_out, NANOARROW_TYPE_STRUCT));
     NANOARROW_RETURN_NOT_OK(ArrowArrayAllocateChildren(array_out, static_cast<int64_t>(n_items)));
-
     array_out->length = 1;
-    array_out->null_count = -1;
```
Comment (josevalim) [inline c_src/adbc_nif.cpp:1258]: This was the culprit. :)

**Pull request merged**